### DISCUSSION TITLE
Javascript Judge in Dodona -> internal error

### DISCUSSION THREAD

**Discussion by RobbeW** [dodona-edu/feedback-and-support#13] [Questions - Vragen] [answered]
Hi all, 

When using Dodona to build a set of exercises on Javascript, I can' seem to get the judge to respond correctly. I have tried adding a reference to an testsuite.json as described in the TESTed documentation, but that seems to trigger the same internal error. 

The internal error in question: 
```
Judge exited with status code 1.
Standard Error:
node:internal/fs/utils:344
    throw err;
    ^

Error: ENOENT: no such file or directory, open '/mnt/z9UXMFN1LDKj7fmsN7rN7w/resources/tests.js'
    at Object.openSync (node:fs:585:3)
    at Object.readFileSync (node:fs:453:35)
    at TestParser.parse (/mnt/z9UXMFN1LDKj7fmsN7rN7w/judge/test-parser.js:52:16)
    at new Judge (/mnt/z9UXMFN1LDKj7fmsN7rN7w/judge/judge.js:112:42)
    at Socket.<anonymous> (/mnt/z9UXMFN1LDKj7fmsN7rN7w/judge/evaluate.js:19:19)
    at Socket.emit (node:events:520:28)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: '/mnt/z9UXMFN1LDKj7fmsN7rN7w/resources/tests.js'
}
/main.sh: 1: pkill: not found
Standard Output: 
```
Suggestion on what I appear to be doing incorrectly? 

Thanks in advance!

**Comment by chvp** [comment, marked as answer]
I think there is some confusion about the TESTed judge vs. the JavaScript judge. The JavaScript judge is a judge created for just JavaScript, and expects a `tests.js` file in the `evaluation` directory. The TESTed judge is a new judge supporting multiple programming languages. This is the one expecting the `testsuite.json` file in the `evaluation` directory.

**Reply by RobbeW** [reply]
Thanks for the reply. 
If the preliminary goal here is just 'getting syntax feedback' or 'no internal error' to the students (no full check on their input code); what would Dodona require? 

(I only have some experience using the Python / Pythia judge to give simple syntax feedback by not adding  a specified path or testplan; but Javascript does not seem to trigger the same behaviour)

**Reply by chvp** [reply]
This is unfortunately not possible in the JavaScript judge. It might be possible with TESTed, unfortunately I can't check that as easily (and I'm supervising an exam right now). Maybe @niknetniko knows by heart whether this is possible with TESTed?

**Reply by niknetniko** [reply]
I've looked into it; TESTed supports this (at least for JavaScript). An example is https://github.com/niknetniko/example-exercises/tree/master/syntax-check.

You need to have an empty object in [`testsuite.json`](https://github.com/niknetniko/example-exercises/blob/master/syntax-check/evaluation/empty.tson):

```json
{}
```

It is also important to set the following option in the [`config.json`](https://github.com/niknetniko/example-exercises/blob/e1fb7bf8e5efb513b0869e1cebe06ff8a74f04d6/syntax-check/config.json#L14) file:

```json
{
  "options": {
    "allow_fallback": false
  }
}
```

**Reply by RobbeW** [reply]
@niknetniko this seems to do the trick. 
I owe you a fancy coffee! Thanks for the swift reply / help!
